addons/addon-base-raas/packages/base-raas-services/lib/aws-accounts/aws-cfn-service.js

```diff
@@ -212,19 +228,40 @@ class AwsCfnService extends Service {
       { action: 'check-aws-permissions', conditions: [allowIfActive, allowIfAdmin] },
       { accountId },
     );
+
     const awsAccountsService = await this.service('awsAccountsService');
     const accountEntity = await awsAccountsService.mustFind(requestContext, { id: accountId });
 
+    // Special case handling for accounts upgrading to this feature
+    if (accountEntity.cfnStackName === '' || accountEntity.cfnStackName === undefined) {
+      return 'UNKNOWN';
+    }
+
     const [cfnTemplateService] = await this.service(['cfnTemplateService']);
     const expectedTemplate = await cfnTemplateService.getTemplate('onboard-account');
 
-    // whitespace and comments removed before comparison
-    const curPermissions = await this.getStackTemplate(requestContext, accountEntity);
-    const trimmedCurPermString = curPermissions.replace(/#.*/g, '').replace(/\s+/g, '');
-    const trimmedExpPermString = expectedTemplate.replace(/#.*/g, '').replace(/\s+/g, '');
+    const stackTemplate = await this.getStackTemplate(requestContext, accountEntity);
+    const stackStatus = stackTemplate.stackStatus;
 
-    // still hash values
-    return trimmedExpPermString !== trimmedCurPermString ? 'NEEDSUPDATE' : 'CURRENT';
+    // some statuses we can determine based on stack status
+    if (PENDING_STATES.includes(stackStatus)) {
+      return 'PENDING';
+    }
+    if (FAILED_STATES.includes(stackStatus)) {
+      return 'ERRORED';
+    }
+    if (COMPLETE_STATES.includes(stackStatus)) {
+      const curPermissions = stackTemplate.permString;
+      // whitespace and comments removed before comparison
+      const trimmedCurPermString = curPermissions.replace(/#.*/g, '').replace(/\s+/g, '');
+      const trimmedExpPermString = expectedTemplate.replace(/#.*/g, '').replace(/\s+/g, '');
+
+      // still hash values
+      return trimmedExpPermString !== trimmedCurPermString ? 'NEEDS_UPDATE' : 'CURRENT';
+    }
+
+    // we should never make it here unless something really goes wrong
+    return 'UNKNOWN';
   }
 
   async batchCheckAccountPermissions(requestContext, batchSize = 5) {
```

Comment:
should this be `batchCheckAndUpdateAccountPermissions`?

Reply:
uhhhhh.....it probably be should be haha, at first it was just checking but now it's updating--I'll change it everywhere it appears